src/components/LegendPanel.vue

```diff
@@ -1,87 +1,206 @@
 <template>
-  <aside class="legend-panel">
-    <h2 class="legend-panel__title">Legenda</h2>
-    <figure v-for="layer in formattedLayers" :key="layer.id" class="legend-panel__legend" v-test="'legend'">
-      <template v-if="layer.legend">
-        <figcaption class="legend-panel__caption">{{ layer.legend.title }}</figcaption>
-        <slot></slot>
-        <!-- lookup legend if slot is empty -->
-        <img :src="layer.legendImageSrc" alt="">
-      </template>
-    </figure>
+  <aside
+    :class="{
+      'legend-panel': true,
+      'legend-panel--active': legendIsShown,
+    }"
+    @mouseenter="showLegend"
+    @mouseleave="hideLegend"
+  >
+    <button
+      class="legend-panel__title"
+      aria-controls="legend"
+      :aria-expanded="legendIsShown"
+      @click="toggleLegend"
+    >
+      Legenda
+      <img
+        v-if="formattedLayers.length > 1"
+        class="legend-panel__collapse-icon"
+        :src="`${publicPath}icons/baseline-keyboard_arrow_up-24px.svg`"
+        :alt="`Klap legenda ${legendIsShown ? 'in' : 'uit'}`"
+      />
```

Comment:
Ik zou deze image role=presentation geven en de button een aria-label geven met `Klap legenda ${legendIsShown ? 'in' : 'uit'}`